### PR TITLE
Undo and set bugfixes, and entity handling

### LINKED ISSUE #50: Undo doesn't update block states
This probably extends to other (or all) setting functions, since the cause of the bug is [this line](https://github.com/Ghoulboy78/World-edit-scarpet/blob/76e6e23cbf2b93a8cbcf920c8e1ba051ea31310e/world-edit.sc#L1316), where it checks if the new block batches the old one, in which case it doesn't set, but it doesn't check for different block states.

I noticed this when removing lava and then undoing the action, which left flowing lava, because `undo` didn't replace flowing lava with lava sources (both are lava blocks). I haven't done further testing, but it likely affects the whole app.

### LINKED ISSUE #57: Rotate doesn't behave properly with undo
For some strange reason, rotating a big structure some angle that is not multiple of 90 degrees and the undoing it leaves blocks behind. I didn't do any code analysis to check why that's the case.

Result of rotating an octahedron (into a space that was formerly just air)
image

After undoing the operation
image

### PR DESCRIPTION
**Pull request by Firigion**
Fixes #57 and fixes #50 with the solutions proposed in each bugreport.

### CODE REVIEW

**Comment by Ghoulboy78** [comment]
hmm, your fix for the undo now makes it so if I set a block with the same state as before, it still thinks that many blocks changed.

**Comment by Firigion** [comment]
Woops, I forgot to check that. I'll see what's up.

**Review by altrisi — COMMENTED** [review]
Other than that and what Ghoulboy said, fixes the bugs and looks good.

Seems like anything gets added to the changed list, not only same blockstate.

File: se.sc [hunk 1]
```diff
@@ -44,7 +44,8 @@ base_commands_map = [
     ['angel', '_give_angel_block_item', [-1, 'help_cmd_angel_give', null, null]],
     ['angel new', '_set_angel_block_item', [-1, 'help_cmd_angel_new', null, null]],
     ['angel clear', '_clear_angel_block_item', [-1, 'help_cmd_angel_clear', null, null]],
-    ['rotate <pos> <degrees> <axis>', 'rotate', [-1, 'help_cmd_rotate', 'help_cmd_rotate_tooltip', null]],//will replace old stuff if need be
+    ['rotate <pos> <degrees> <axis>', ['rotate', null], [-1, 'help_cmd_rotate', 'help_cmd_rotate_tooltip', null]],//will replace old stuff if need be
+    ['rotate <pos> <degrees> <axis> f <flag>', 'rotate', [-1, 'help_cmd_rotate', 'help_cmd_rotate_tooltip', null]],//will replace old stuff if need be
```
[the remark this answers is not in the document]
Comment (Firigion) [reply se.sc:48]: Yes, I missed that one, thanks!

**Comment by Firigion** [comment]
I need to be more moderate and make multiple PRs. Sorry, people :/

**Review by altrisi — APPROVED** [review]

**Comment by altrisi** [comment]
Did the minimum review count change? (should I merge it?)

**Comment by Firigion** [comment]
It changed when the activity here lowered and fewer people were reviewing stuff. Two reviews meant that the whole active team had to agree, since there were only three of us ding stuff :P Idk if it was the right decission, but it was the only way to keep developing the app in the meantime.
I don't recall this having any bugs, so it's good to merge, as far as I'm concerned. Idk if you tested or not tho...

**Comment by altrisi** [comment]
Yes I tested it back in my previous review and again today. I'll merge it, if it breaks, it can be fixed SoonTM.

**Pull request merged**